src/gui/properties/speedplotview.h

```diff
@@ -0,0 +1,89 @@
+#ifndef SPEEDPLOTVIEW_H
+#define SPEEDPLOTVIEW_H
+
+#include <QGraphicsView>
+#include <QPen>
+#include <QQueue>
+#include <QMap>
+
```

Comment:
>but only for fix circular dependency errors.

Just so you know another reason it is useful: For a project with many .h/.cpp files, forward declaration reduces the time needed for compilation. The compiler doesn't need to open the included headers for every place YOUR header gets included in YOUR code.
Headers are small files but filesystem **access** is tremendously slow compared to the speed our CPUs can compile stuff.